configure

```diff
@@ -257,33 +257,54 @@ touch s.h m.h Makefile
 
 # Write options to Makefile
 
+config() {
+# This function hardens the generated Makefile against '#' symbols
+# present in a source path (opam-compiler-conf may pick such directory
+# names if working from a branch named 'PR#4242-answer-all-questions')
+# by escaping them into '\#'.
+
+# When injecting data in Makefiles, it is customary to also escape
+# '$', which get turned into '$$'. However, this transformation is
+# invalid here as some of the variables are meant to be code
+# interpreted by make: for example, passing
+#   --bindir "$(PREFIX)/bin2"
+# is explicitly supported (see "or relative to $(PREFIX)" messages above).
+
+# Finally, it is also impossible for the user to escape the '#' signs
+# before calling this configure script, given that
+# $(PREFIX) is also injected in C code where this escape is invalid
+# -- see the definition of the OCAML_STDLIB_DIR macro below.
+
+  echo $1=$2 | sed 's/\#/\\#/g' >> Makefile
```

Comment:
`NATIVECCLIBS` gets two spaces because `$mathlib` is empty (comes via `$cclibs`); `BYTECCLIBS` gets 18 spaces because of the "bad" line continuation. If it's changed to:

```bash
echo "BYTECCLIBS=$cclibs $dllib $curseslibs $pthread_link" \
                "$instrumented_runtime_libs" >> Makefile
```
then it should work, however, none of them causes a problem as all those spaces get compacted when make loads config/Makefile so I'm not sure I'd change any of it (as much as I like perfectly generated files!)